pool.go

```diff
@@ -395,8 +428,14 @@ func (p *Pool[T]) acquire(ctx context.Context) (*Resource[T], error) {
 
 	res, err := p.initResourceValue(ctx, res)
 	if err != nil {
+		defer trace.acquireEndErr(ctx, err)
 		return nil, err
 	}
+	if waitedForLock {
+		defer trace.acquireEnd(ctx, time.Duration(nanotime()-startNano), res.traceStats, true)
+	} else {
+		defer trace.acquireEnd(ctx, 0, res.traceStats, true)
+	}
```

Comment:
This is similar to the TODO above. Ideally, `waitTime` would be declared, but unset until the check of `waitedForLock`. Then there would only be two calls to `trace.acquireEnd`, one for pooled values and one for new ones.